compiler/compile/OMRCompilation.cpp

```diff
@@ -2730,3 +2730,18 @@ void OMR::Compilation::invalidateAliasRegion()
       new (&self()->_aliasRegion) TR::Region(_heapMemoryRegion);
       }
    }
+
+/*
+ * \brief
+ *    This query tells whether the trees might contain rdbar/wrtbar opcodes
+ *
+ * \note
+ *    This query is for temporarily disable opts not supporting
+ *    the newly added rdbar/wrtbar opcodes. Subprojects can overrite
+ *    the answer. This query should be deleted eventually if
+ *    all the optimizations support the opcodes.
+ */
+bool OMR::Compilation::mayContainUnsupportedReadbarOrWritebar()
```

Comment:
I was hoping to avoid confusion with the existing `awrtbari` and`awrtbar`. Those wrtbars are supported by all the opts in context of GC. `mayContainReadbarOrWritebar` sounds like it's referring to all read and write bars in all cases.

Reply:
My objection is around the generality of this query.  "unsupported" leaves one to wonder where and how they are unsupported.  I think if the context could be more clearly defined then it would be ok.  For example, if this is to prevent optimizations from running that may not competely implement support for all barriers then the somewhat wordy `incompleteOptimizerSupportForReadWriteBarriers` might be a less ambiguous name for this query.

Ideally, we should prevent optimizations from even running if we generate read/write barriers during IL gen that we know have incomplete support in certain optimizations, but we don't (yet!) have that capability.

Reply:
Changed the function name as suggested to be more clear in the intention.